File: 03_Arreglos/Vectores/Python/ejemplo2.py
def busqueda_lineal(v, x):
    for i, val in enumerate(v):
        if val == x:
            return i
    return -1

def main():
    n = int(input("Ingrese N (tamaño del vector): "))
    v = [int(input(f"v[{i}] = ")) for i in range(n)]

    if n == 0:
        print("Vector vacío.")
        return

    maximo, idx = v[0], 0
    for i in range(1, n):
        if v[i] > maximo:
            maximo, idx = v[i], i

    print(f"\nMáximo = {maximo} en la posición {idx}")

    x = int(input("\nIngrese valor a buscar x: "))
    pos = busqueda_lineal(v, x)
    print(f"Encontrado en posición {pos}" if pos >= 0 else "No encontrado (retorna -1)")

if __name__ == "__main__":
    main()

File: 03_Arreglos/Vectores/CSharp/ejemplo2.cs
using System;
class Programa {
    static void Main() {
        Console.Write("Ingrese N (tamaño del vector): ");
        int n = int.Parse(Console.ReadLine() ?? "0");
        int[] v = new int[n];

        for (int i = 0; i < n; i++) {
            Console.Write($"v[{i}] = ");
            v[i] = int.Parse(Console.ReadLine() ?? "0");
        }

        if (n == 0) { Console.WriteLine("Vector vacío."); return; }

        int max = v[0], idx = 0;
        for (int i = 1; i < n; i++)
            if (v[i] > max) { max = v[i]; idx = i; }

        Console.WriteLine($"\nMáximo = {max} en la posición {idx}");

        Console.Write("\nIngrese valor a buscar x: ");
        int x = int.Parse(Console.ReadLine() ?? "0");
        int pos = -1;
        for (int i = 0; i < n; i++)
            if (v[i] == x) { pos = i; break; }

        Console.WriteLine(pos >= 0 ? $"Encontrado en posición {pos}" : "No encontrado (retorna -1)");
    }
}
